src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Type/TypeTests.cs

```diff
@@ -509,6 +509,26 @@ public void GetTypeByName_ValidType_ReturnsExpected(string typeName, Type expect
             Assert.Equal(expectedType, Type.GetType(typeName.ToLower(), throwOnError: false, ignoreCase: true));
         }
 
+        public static IEnumerable<object[]> GetTypeByName_InvalidElementType()
+        {
+            Type expectedException = PlatformDetection.IsMonoRuntime
+                ? typeof(ArgumentException) // https://github.com/dotnet/runtime/issues/45033
```

Comment:
This is reintroducing bugs and behavior differences since Mono is not using the shared managed implementation of type name parsing anymore in the affected APIs.